src/suricata.c

```diff
@@ -2800,8 +2801,10 @@ int SuricataMain(int argc, char **argv)
     SC_ATOMIC_SET(engine_stage, SURICATA_RUNTIME);
     PacketPoolPostRunmodes();
 
-    /* Un-pause all the paused threads */
+    /* Un-pause all the paused threads, and wait for them to start */
     TmThreadContinueThreads();
+    TmThreadEnsureUnpaused();
+    TmThreadEnsureRunning();
```

Comment:
this adds 2 loops that (potentially) sleep, how does this affect start up time? I fear it can add quite a bit of time for runs with many cores.

Reply:
Hmm... I can look into the affect it has.

I think it shouldn't actually affect packet processing, as the threads themselves continue running as usual, this just creates a sync point, to ensure all the threads are up and running, before executing the remaining calls (PostRunStartedDetectSetup(), SCPledge() and SuricataMainLoop()), which largely just setup and monitor for signals, I believe, right?

That said; I feel like this is mostly useful in an integration/end-to-end test scenario, so maybe this sync point should be command line configurable?  So, in production, you can bypass it if you want.

What do you think?